app/src/test/java/com/github/se/eventradar/EventOverviewViewModelTest.kt

```diff
@@ -0,0 +1,189 @@
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.event.MockEventRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import com.github.se.eventradar.model.repository.user.MockUserRepository
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.mockkStatic
+import io.mockk.verify
+import java.time.LocalDateTime
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestDispatcher
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+@ExperimentalCoroutinesApi
+class EventsOverviewViewModelTest {
+
+  private lateinit var viewModel: EventsOverviewViewModel
+  private lateinit var eventRepository: IEventRepository
+  private lateinit var userRepository: IUserRepository
+
+  class MainDispatcherRule(
+      private val testDispatcher: TestDispatcher = UnconfinedTestDispatcher()
+  ) : TestWatcher() {
+    override fun starting(description: Description) {
+      Dispatchers.setMain(testDispatcher)
+    }
+
+    override fun finished(description: Description) {
+      Dispatchers.resetMain()
+    }
+  }
+
+  @get:Rule val mainDispatcherRule = MainDispatcherRule()
+
+  private val mockEvent =
+      Event(
+          eventName = "Event 1",
+          eventPhoto = "",
+          start = LocalDateTime.now(),
+          end = LocalDateTime.now(),
+          location = Location(0.0, 0.0, "Test Location"),
+          description = "Test Description",
+          ticket = EventTicket("Test Ticket", 0.0, 1),
+          contact = "Test Contact Email",
+          organiserList = setOf("Test Organiser"),
+          attendeeList = setOf("Test Attendee"),
+          category = EventCategory.COMMUNITY,
+          fireBaseID = "1")
+
+  private val mockUser =
+      User(
+          userId = "user1",
+          age = 30,
+          email = "test@example.com",
+          firstName = "John",
+          lastName = "Doe",
+          phoneNumber = "1234567890",
+          accountStatus = "active",
+          eventsAttendeeList = listOf("1", "2"),
+          eventsHostList = listOf("3"),
+          profilePicUrl = "http://example.com/pic.jpg",
+          qrCodeUrl = "http://example.com/qr.jpg",
+          username = "john_doe")
+
+  @Before
+  fun setUp() {
+    eventRepository = MockEventRepository()
+    userRepository = MockUserRepository()
+    viewModel = EventsOverviewViewModel(eventRepository, userRepository)
+  }
+
+  @Test
+  fun testGetEventsEmpty() = runTest {
+    viewModel.getEvents()
+
+    advanceUntilIdle()
```

Comment:
nit: I don't believe this is necessary. This is handled automatically by `runTest`.